keyinfo.go

```diff
@@ -0,0 +1,114 @@
+package parsec
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/parallaxsecond/parsec-client-go/algorithm"
+	"github.com/parallaxsecond/parsec-client-go/wireinterface/operations/listkeys"
+	"github.com/parallaxsecond/parsec-client-go/wireinterface/operations/psaalgorithm"
+	"github.com/parallaxsecond/parsec-client-go/wireinterface/operations/psakeyattributes"
+)
+
+type ToWire interface {
+	ToWireInterface() interface{}
+}
+
+type UsageFlags struct {
+	Export        bool
+	Copy          bool
+	Cache         bool
+	Encrypt       bool
+	Decrypt       bool
+	SignMessage   bool
+	VerifyMessage bool
+	SignHash      bool
+	VerifyHash    bool
+	Derive        bool
+}
+
+func (u *UsageFlags) toNativeWireInterface() *psakeyattributes.UsageFlags {
+	return &psakeyattributes.UsageFlags{
+		Export:        u.Export,
+		Copy:          u.Copy,
+		Cache:         u.Cache,
+		Encrypt:       u.Encrypt,
+		Decrypt:       u.Decrypt,
+		SignMessage:   u.SignMessage,
+		VerifyMessage: u.VerifyMessage,
+		VerifyHash:    u.VerifyHash,
+		SignHash:      u.SignHash,
+		Derive:        u.Derive,
+	}
+}
+
+type KeyPolicy struct {
+	KeyUsageFlags *UsageFlags
+	KeyAlgorithm  *algorithm.Algorithm
+}
+
+func (kp *KeyPolicy) toNativeWireInterface() *psakeyattributes.KeyPolicy {
+	kaif := kp.KeyAlgorithm.ToWireInterface()
+	if kaif == nil {
+		fmt.Println("no wire alg from kp.KeyAlgorithm")
+		return nil
+	}
+	ka, ok := kaif.(*psaalgorithm.Algorithm)
+	if !ok {
+		fmt.Printf("kaif is wrong type, got %v\n", reflect.TypeOf(kaif))
+		return nil
+	}
+
+	return &psakeyattributes.KeyPolicy{
+		KeyAlgorithm:  ka,
+		KeyUsageFlags: kp.KeyUsageFlags.toNativeWireInterface(),
+	}
+}
+
+type KeyAttributes struct {
+	KeyType   *KeyType
+	KeyBits   uint32
+	KeyPolicy *KeyPolicy
+}
+
+func newKeyAttributesFromOp(ka *psakeyattributes.KeyAttributes) (*KeyAttributes, error) { //nolint:unparam
+	return &KeyAttributes{
+		KeyBits: ka.KeyBits,
+		// KeyPolicy: ,
+	}, nil
+	// TODO finish this
```

Comment:
I'd say leave this as is and it can be expanded/improved in other PRs! Similarly for other TODOs or FIXMEs